sklearn/preprocessing/tests/test_common.py

```diff
@@ -113,29 +129,20 @@ def test_missing_value_handling(
             Xt_dense = est_dense.fit(X_train).transform(X_test)
             Xt_inv_dense = est_dense.inverse_transform(Xt_dense)
 
-        for sparse_constructor in (
-            sparse.csr_matrix,
-            sparse.csc_matrix,
-            sparse.bsr_matrix,
-            sparse.coo_matrix,
-            sparse.dia_matrix,
-            sparse.dok_matrix,
-            sparse.lil_matrix,
-        ):
             # check that the dense and sparse inputs lead to the same results
             # precompute the matrix to avoid catching side warnings
-            X_train_sp = sparse_constructor(X_train)
-            X_test_sp = sparse_constructor(X_test)
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore", PendingDeprecationWarning)
-                warnings.simplefilter("error", RuntimeWarning)
-                Xt_sp = est_sparse.fit(X_train_sp).transform(X_test_sp)
-
-            assert_allclose(Xt_sp.A, Xt_dense)
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore", PendingDeprecationWarning)
-                warnings.simplefilter("error", RuntimeWarning)
-                Xt_inv_sp = est_sparse.inverse_transform(Xt_sp)
+        X_train_sp = sparse_container(X_train)
+        X_test_sp = sparse_container(X_test)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PendingDeprecationWarning)
+            warnings.simplefilter("error", RuntimeWarning)
+            Xt_sp = est_sparse.fit(X_train_sp).transform(X_test_sp)
+
+        assert_allclose(Xt_sp.A, Xt_dense)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PendingDeprecationWarning)
+            warnings.simplefilter("error", RuntimeWarning)
+            Xt_inv_sp = est_sparse.inverse_transform(Xt_sp)
 
             assert_allclose(Xt_inv_sp.A, Xt_inv_dense)
```

Comment:
```suggestion
            assert_allclose(Xt_inv_sp.toarray(), Xt_inv_dense)
```

Reply:
made changes. Let's see how the CI will react to it .